server/src/main/kotlin/fi/oph/kitu/logging/OpenTelemetryExtensions.kt

```diff
@@ -0,0 +1,14 @@
+package fi.oph.kitu.logging
+
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.context.Scope
+
+inline fun <T> Span.use(block: (Span) -> T): T {
+    val scope: Scope = this.makeCurrent()
+    return try {
+        block(this)
+    } finally {
```

Comment:
Tähän voisi lisätä vielä catch-blokin joka kutsuu recordExceptionia, jolloin spanit jotka heittää poikkeuksen merkitään virheellisiksi: `this.recordException(ex)`.

Reply:
(Nämä ovat kaksi erillistä namiskaa varmaan siksi, että joskus voi haluta tallentaa spanin tietoihin poikkeuksen, mutta jos tietää että tämä poikkeus ei johda sovelluksen virhetilanteeseen voi olla kutsumatta setStatusta.)